SE-20362: the mule.wmq.encodingCorrelationId.disable system property … (#2011)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/mule-wmq-transport-reference.adoc

The example that ships with the Mule WMQ transport allows you to test retry policies. For complete information, see the readme file in the WMQ distribution.

[[using-correlationid-provided]]

> == Using a Correlation ID Provided

Reviewer: ```suggestion
== Using a Correlation ID
```

When an application invokes a request-response operation, you can use a correlation ID to correlate response messages with request messages. With WebSphere® MQ and WebSphere MQ JMS, you can correlate a response message using either a correlation ID or a message ID. In most cases, the caller lets the queue manager select a message ID and expects the application to copy the message ID into the correlation ID of the response message.

> The JMSCorrelationID can be a byte[] value, a string value containing hexadecimal characters and prefixed with `ID:`, or an arbitrary string value not beginning `ID:`. By default, the Mule correlation ID (or any correlation ID provided) sent to WMQ in a message is encoded with the WebSphere format.

Reviewer: ```suggestion
The JMSCorrelationID can be a byte[] value, a string value containing hexadecimal characters and prefixed with `ID:`, or an arbitrary string value that does not begin with `ID:`. By default, the Mule correlation ID (or any correlation ID provided) is sent to WMQ in a message that is encoded with the WebSphere format.
```

> To preserve the correlation ID provided and avoid encoding, set the `mule.wmq.encodingCorrelationId.disable` system property to `true`:

Reviewer: ```suggestion
To preserve the provided correlation ID and avoid encoding, set the `mule.wmq.encodingCorrelationId.disable` system property to `true`:
```

*Type*: boolean +
*Required*: no +
*Default*: none

> |correlationId |A client can use the correlation ID header field to link one message to another. A typical use case is to link a response message with its request message. WMQ uses a 24-byte string for correlation ID. WebSphere pads shorter values with zeroes so that the fixed length is always 24 bytes. Because each message sent by a WMQ provider is assigned a message ID value, it is convenient to link messages via the message ID. By default, all message ID values start with the 'ID:' prefix. To preserve a provided correlation ID, review the <<using-correlationid-provided,using a correlation ID provided>> documentation section..

Reviewer: ```suggestion
|correlationId |A client can use the correlation ID header field to link one message to another. A typical use case is to link a response message with its request message. WMQ uses a 24-byte string for the correlation ID. WebSphere pads shorter values with zeroes so that the fixed length is always 24 bytes. Because each message sent by a WMQ provider is assigned a message ID value, it is convenient to link messages via the message ID. By default, all message ID values start with the `ID:` prefix. To preserve a provided correlation ID, review the <<using-correlationid-provided,using a correlation ID provided>> documentation section..
```
